pkg/health/listener.go

```diff
@@ -2,25 +2,22 @@ package health
 
 import (
 	"context"
+	"encoding/json"
 	"math"
 	"math/rand"
-	"os"
-	"strings"
 	"sync"
 	"time"
 
+	"github.com/nats-io/nats.go"
 	agentv1 "github.com/rancher/opni/pkg/agent"
-	"github.com/rancher/opni/pkg/alerting"
-	"github.com/rancher/opni/pkg/alerting/condition"
 	"github.com/rancher/opni/pkg/alerting/shared"
-	alertingv1alpha "github.com/rancher/opni/pkg/apis/alerting/v1alpha"
 	controlv1 "github.com/rancher/opni/pkg/apis/control/v1"
 	corev1 "github.com/rancher/opni/pkg/apis/core/v1"
 	"github.com/rancher/opni/pkg/auth/cluster"
 	"github.com/rancher/opni/pkg/util"
 	"golang.org/x/sync/semaphore"
-	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/emptypb"
+	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
```

Comment:
Instead of writing events to jetstream here, use the Monitor inside the gateway to watch status updates, and move the jetstream-specific implementation to the alerting package

Reply:
where is the API for that one

Reply:
Here https://github.com/rancher/opni/blob/4ca9b29f98099524bb015b4db2f519c98f4056d7/pkg/health/monitor.go#L124
Example usage here: https://github.com/rancher/opni/blob/5f9662f09332a159caccea18476a9e3940717318/pkg/gateway/gateway.go#L331

Reply:
Done